internal/engine/upper_test.go

```diff
@@ -1786,6 +1786,43 @@ func TestDockerComposeFiltersRunLogs(t *testing.T) {
 	})
 }
 
+func TestDockerComposeDetectsCrashes(t *testing.T) {
+	f := newTestFixture(t)
+	m1, m2 := f.setupDCFixture()
+
+	f.loadAndStart()
+	f.waitForCompletedBuildCount(2)
+
+	f.withManifestState(m1.ManifestName().String(), func(st store.ManifestState) {
+		assert.NotEqual(t, dockercompose.StatusCrash, st.DCResourceState().Status)
+	})
+
+	f.withManifestState(m2.ManifestName().String(), func(st store.ManifestState) {
+		assert.NotEqual(t, dockercompose.StatusCrash, st.DCResourceState().Status)
+	})
+
+	f.dcc.SendEvent(dcContainerEvtForManifest(m1, dockercompose.ActionKill))
```

Comment:
i might also test sending an "all is well" sequence of events and check that we're NOT crashing